testsuites/syncgateway/functional/tests/test_longpoll.py

```diff
@@ -131,3 +139,747 @@ def test_longpoll_changes_sanity(params_from_base_test_setup, sg_conf_name, num_
+
+    log_info("sg_conf: {}".format(sg_conf))
+    log_info("sg_admin_url: {}".format(sg_admin_url))
+    log_info("sg_url: {}".format(sg_url))
+
+    cluster = Cluster(config=cluster_conf)
+    mode = cluster.reset(sg_config_path=sg_conf)
+
+    adam_user_info = UserInfo(name="adam", password="Adampass1", channels=["NBC"], roles=[])
+    traun_user_info = UserInfo(name="traun", password="Traunpass1", channels=["CBS"], roles=[])
+    andy_user_info = UserInfo(name="andy", password="Andypass1", channels=["MTV"], roles=[])
+    sg_db = "db"
+
+    client = MobileRestClient()
+
+    adam_auth = client.create_user(url=sg_admin_url, db=sg_db,
+                                   name=adam_user_info.name, password=adam_user_info.password, channels=adam_user_info.channels)
+
+    traun_auth = client.create_user(url=sg_admin_url, db=sg_db,
+                                    name=traun_user_info.name, password=traun_user_info.password, channels=traun_user_info.channels)
+
+    andy_auth = client.create_user(url=sg_admin_url, db=sg_db,
+                                   name=andy_user_info.name, password=andy_user_info.password, channels=andy_user_info.channels)
+
+    # Get starting sequence of docs, use the last seq to progress past any user docs
+    adam_changes = client.get_changes(url=sg_url, db=sg_db, since=0, timeout=30, auth=adam_auth)
+    traun_changes = client.get_changes(url=sg_url, db=sg_db, since=0, timeout=30, auth=traun_auth)
+    andy_changes = client.get_changes(url=sg_url, db=sg_db, since=0, timeout=30, auth=andy_auth)
+
+    with concurrent.futures.ProcessPoolExecutor() as ex:
+        adam_changes_task = ex.submit(client.get_changes, url=sg_url, db=sg_db, since=adam_changes["last_seq"], timeout=30, auth=adam_auth)
+        traun_changes_task = ex.submit(client.get_changes, url=sg_url, db=sg_db, since=traun_changes["last_seq"], timeout=30, auth=traun_auth)
+        andy_changes_task = ex.submit(client.get_changes, url=sg_url, db=sg_db, since=andy_changes["last_seq"], timeout=30, auth=andy_auth)
+
+        # make sure longpoll goes to a wait state
+        log_info("Sleeping (2 sec) to allow longpoll to enter waiting state ...")
+        time.sleep(2)
+
+        # Add one doc, this should wake up the changes feed
+        adam_add_docs_task = ex.submit(client.add_docs, url=sg_url, db=sg_db,
+                                       number=1, id_prefix="adam_doc",
+                                       auth=adam_auth, channels=adam_user_info.channels)
+
+        traun_add_docs_task = ex.submit(client.add_docs, url=sg_url, db=sg_db,
+                                        number=1, id_prefix="traun_doc",
+                                        auth=traun_auth, channels=traun_user_info.channels)
+
+        andy_add_docs_task = ex.submit(client.add_docs, url=sg_url, db=sg_db,
+                                       number=1, id_prefix="andy_doc",
+                                       auth=andy_auth, channels=andy_user_info.channels)
+
```

Comment:
Having the user with the active changes feed also be the user creating the doc that wakes up the feed feels like it might be hiding potential issues.  I think it would be a useful change to use adam_auth to create all three docs - that would cover both scenarios (user-created update, non-user-created update).